File: javascript/fizzbuzz.js
/**
   fizzbuzz.js
   ================================================================
   CREATED: 2018-05-11
   VERSION: 1.0
   AUTHOR : pinecone
   ABOUT  : The FizzBuzz Challenge
   NOTES  : given a number n, return 'fizz' if n is divisible by 3,
            return 'buzz' if n is divisible by 5, return 'fizzbuzz'
            if n  is divisible by both  3 and 5,  or n if n  is not
            divisible by either.
   ----------------------------------------------------------------
 */

const { checkData } = require('../check-data');

const fizzbuzz = n => {
  const fizz = !(n%3),
	buzz = !(n%5);
  return (
    fizz && buzz ? 'fizzbuzz' :
      fizz || buzz ? fizz ? 'fizz' : 'buzz' :
    n
  );
}
 
const data = [
  {in: 0, out: 'fizzbuzz'},
  {in: 1, out: 1},
  {in: 2, out: 2},
  {in: 3, out: 'fizz'},
  {in: 4, out: 4},
  {in: 5, out: 'buzz'},
  {in: 6, out: 'fizz'},
  {in: 7, out: 7},
  {in: 8, out: 8},
  {in: 9, out: 'fizz'},
  {in: 10, out: 'buzz'},
  {in: 11, out: 11},
  {in: 12, out: 'fizz'},
  {in: 13, out: 13},
  {in: 14, out: 14},
  {in: 15, out: 'fizzbuzz'},
  {in: 16, out: 16},
  {in: 17, out: 17},
  {in: 18, out: 'fizz'},
  {in: 19, out: 19},
  {in: 20, out: 'buzz'},
  {in: 21, out: 'fizz'},
  {in: 22, out: 22},
  {in: 23, out: 23},
  {in: 24, out: 'fizz'},
  {in: 25, out: 'buzz'},
  {in: 26, out: 26},
  {in: 27, out: 'fizz'},
  {in: 28, out: 28},
  {in: 29, out: 29},
  {in: 30, out: 'fizzbuzz'},
];

checkData(data, fizzbuzz);

File: challenges/fizzbuzz/ruby/fizzbuzz.rb
#!/usr/bin/env ruby
# fizzbuzz.rb
# =============================================================================
# CREATED: 2018-10-22
# UDPATED: 2018-10-23
# VERSION: 1.2.0
# AUTHOR : wlharvey4
# ABOUT  : FIZZBUZZ implemented in Ruby2_5
# USAGE  : fizzbuzz.rb <int> or Fizzbuzz.new({"n" => <int>})
# NOTES  :
# CHANGE-LOG:
# .............................................................................
# version 1.0.0 2018-10-22T11:30
# -- initial commit
# .............................................................................
# version 1.1.0 2018-10-22T16:00
# -- change input to be a hash instead of json
# -- change symbols to strings for easier equality check
# -- remove option to compare two values from the command line
# .............................................................................
# version 1.2.0 2018-10-23T03:00
# -- changed attr_accessor to attr_reader
# -- added a comment to the '==' method
# -----------------------------------------------------------------------------

# Class Fizzbuzz receives a Ruby Hash object as described below, and calculates
# a Fizzbuzz  value, which can  be observed  through either #puts  or #inspect.
class Fizzbuzz
  attr_reader :n, :out
  # 'input' is a Ruby Hash of the form {"n" => <int>}
  # @n is obtained from the Hash value as a Fixnum
  # @out is calculated as one of <int> | "fizz" | "buzz" | "fizzbuzz"
  def initialize input
    @n    = input["n"]
    @out  = calculate @n
  end

  # this is Fizzbuzz
  def calculate n
    fizz = n % 3 == 0
    buzz = n % 5 == 0
    if fizz && buzz
      @out = "fizzbuzz"
    elsif fizz || buzz
      @out = fizz ? "fizz" : "buzz"
    else
      @out = n
    end
  end

  # expected is a  single value, either an integer, or  a string containing one
  # of "fizz", "buzz", "fizzbuzz"
  def == expected
      self.out == expected
  end

  def to_s
    "#{@out}"
  end

  def inspect
    "Fizzbuzz of #{@n} is #{@out}"
  end
end

# Fizzbuzz  can be  run from  the  command line  as ruby  fizzbuzz.rb <int>  or
# ./fizzbuzz <int> if it has been made executable and linked.
def main n
  fizzbuzz = Fizzbuzz.new ({"n" => n.to_i})
  p fizzbuzz
end

if $0 == __FILE__
  main $*[0]
end

